framework/db/QueryInterface.php

```diff
@@ -252,4 +252,15 @@ public function limit($limit);
      * @return $this the query object itself
      */
     public function offset($offset);
+
+    /**
+     * Cancels the query actual execution, prevention interaction with data storage.
+     * After this method is triggered, methods, returning query results like [[one()]], [[all()]], [[exists()]]
+     * and so on will return empty or false values.
+     * You should use this method in case your program logic indicates query should not return any results, like
+     * in case you set false where condition like `0=1`.
+     * @return $this the query object itself
+     * @since 2.0.11
+     */
+    public function cancel();
```

Comment:
I think the name of the method isn't OK since we as users do not clearly separate actual SQL query and getting the result. I propose using another name such as `skipQuery()` (not sure it's the best one).